### PR TITLE
clean `unflatten` in `PrepSelPrep` 

### PR DESCRIPTION
**Pull request by KetpuntoG**
**Context:**

LinearCombination was being cast within unflaten. It has been moved to init to avoid issues

### CODE REVIEW

**Review by astralcai — APPROVED** [review]
LGTM!

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 * The `qml.PrepSelPrep` template is added. The template implements a block-encoding of a linear 
   combination of unitaries.
-  [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756)
+  [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756), [(#5987)](https://github.com/PennyLaneAI/pennylane/pull/5987)
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:12]:
```suggestion
  [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756)
  [(#5987)](https://github.com/PennyLaneAI/pennylane/pull/5987)
```

**Review by albi3ro — APPROVED** [review]
🎉 Thanks for this fix :)

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -14,6 +14,8 @@
   combination of unitaries.
   [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756)
 
+[(#5987)](https://github.com/PennyLaneAI/pennylane/pull/5987)
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:17]: Merge conflict?

**Comment by KetpuntoG** [comment]
[sc-68516]

File: doc/releases/changelog-dev.md (hunk 2 above)
Comment (KetpuntoG) [reply doc/releases/changelog-dev.md:17]: Oh sorry you are right! let me fix that

**Pull request merged**